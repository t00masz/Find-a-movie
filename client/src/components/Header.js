import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = (props) => (
  <div className="header">
    <div>
      <h1 className="header__title">
        {props.title}
      </h1>
    </div>
    <div>
      <div className="header__content">
        <NavLink to="/" className="option__link" activeClassName="option__link-is-active" exact={true}>
          Search
        </NavLink>  
        <NavLink to="/favourites" className="option__link" activeClassName="option__link-is-active">
          Favourites
        </NavLink>  
        <NavLink to="/comments" className="option__link" activeClassName="option__link-is-active">
          Comments
        </NavLink> 
      </div>
    </div>
  </div>
);

export default Header;
