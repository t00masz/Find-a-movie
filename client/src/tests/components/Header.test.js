import React from 'react';
import { shallow } from 'enzyme'
import Header from '../../components/Header';

test('should render header', () => {
    const wrapper = shallow(<Header 
        title="Contact form" 
        subtitle="Personal data" />);
    expect(wrapper).toMatchSnapshot();
});
