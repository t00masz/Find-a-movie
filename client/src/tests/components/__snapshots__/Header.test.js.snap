// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    subtitle="Personal data"
    title="Contact form"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
>
        <h1
                className="header__title"
        >
                Contact form
        </h1>
        <h2
                className="header__subtitle"
        >
                Personal data
        </h2>
</div>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="header__title"
>
            Contact form
</h1>,
          <h2
            className="header__subtitle"
>
            Personal data
</h2>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Contact form",
            "className": "header__title",
          },
          "ref": null,
          "rendered": "Contact form",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Personal data",
            "className": "header__subtitle",
          },
          "ref": null,
          "rendered": "Personal data",
          "type": "h2",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          <h1
                    className="header__title"
          >
                    Contact form
          </h1>
          <h2
                    className="header__subtitle"
          >
                    Personal data
          </h2>
</div>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="header__title"
>
              Contact form
</h1>,
            <h2
              className="header__subtitle"
>
              Personal data
</h2>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Contact form",
              "className": "header__title",
            },
            "ref": null,
            "rendered": "Contact form",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Personal data",
              "className": "header__subtitle",
            },
            "ref": null,
            "rendered": "Personal data",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
